firestore/src/android/firestore_android.cc

```diff
@@ -515,6 +528,49 @@ void FirestoreInternal::ClearListeners() {
   listener_registrations_.clear();
 }
 
+Future<LoadBundleTaskProgress> FirestoreInternal::LoadBundle(
+    const std::string& bundle) {
+  Env env = GetEnv();
+  Local<String> bundle_jstring = env.NewStringUtf(bundle.c_str());
+  Local<String> encoding = env.NewStringUtf("UTF-8");
+  Local<Array<uint8_t>> bytes = bundle_jstring.GetBytes(env, encoding);
+  Local<LoadBundleTaskInternal> task = env.Call(obj_, kLoadBundle, bytes);
+
+  return promises_->NewFuture<LoadBundleTaskProgress>(env, AsyncFn::kLoadBundle,
+                                                      task);
+}
+
+Future<LoadBundleTaskProgress> FirestoreInternal::LoadBundle(
+    const std::string& bundle,
+    std::function<void(const LoadBundleTaskProgress&)> progress_callback) {
+  Env env = GetEnv();
+  Local<String> bundle_jstring = env.NewStringUtf(bundle.c_str());
+  Local<String> encoding = env.NewStringUtf("UTF-8");
+  Local<Array<uint8_t>> bytes = bundle_jstring.GetBytes(env, encoding);
+  Local<LoadBundleTaskInternal> task = env.Call(obj_, kLoadBundle, bytes.get());
+
+  auto* listener = new LambdaEventListener<LoadBundleTaskProgress>(
```

Comment:
Where does `listener` get deleted? Does the `LambdaEventListener` look after this? I poked around the code quickly and couldn't find where it gets deleted.

Reply:
As discussed offline, I decided to take up some extent of memory leak to support the unlikely case of deleting firestore in the middle of progress update.